ear-production-suite-plugins/plugins/hoa/src/hoa_frontend_connector.cpp

```diff
@@ -195,7 +195,11 @@ void HoaJuceFrontendConnector::trackPropertiesChanged(
 void HoaJuceFrontendConnector::comboBoxChanged(EarComboBox* comboBox) {
   if (!hoaTypeComboBox_.expired() &&
       comboBox == hoaTypeComboBox_.lock().get()) {
```

Comment:
I know this is existing code, but this might be better as
```c++
if(auto thisBox = hoaTypeComboBox_.lock(); thisBox && thisBox.get() == comboBox)  {
\\ ...
}
```
That would ensure thisBox (and so comboBox if the condition is true) will be kept alive for the duration of the if statement.
With the current code it might be possible for whatever holds the shared_ptr to go out of scope between the call to expired() and the call to lock(). It's not an issue if the other end of this is on the same thread, but it's good practice to hold onto a shared ptr for the time you're using it.

Same  pattern for the routing combo box.

Could factor it out into

```c++
namespace {
std::shared_ptr<EarComboBox> matchingSharedPtr(std::weak_ptr<EarComboBox> weak, 
                                               EarComboBox* raw) {
  auto shared = weak.lock();
  if(raw && shared && shared.get() == raw) { 
    return shared; 
  }
  return nullptr;
}
}
```

then just do

```c++
if(auto hoaTypeComboBox = matchingSharedPtr(hoaTypeComboBox_, comboBox))
```
I guess?

Reply:
Good point - we should probably do this for all instances of that pattern in that case. A search through all files for `_.expired() &&` gives 57 matches and they all look like it's that same pattern ☹️

Reply:
OK I've reverted the previous commit and submitted the global weak_ptr changes separately as #181